core/execution/market.go

```diff
@@ -3154,16 +3154,22 @@ func (m *Market) commandLiquidityAuction(ctx context.Context) {
 func (m *Market) tradingTerminated(ctx context.Context, tt bool) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.mkt.State = types.MarketStateTradingTerminated
-	m.mkt.TradingMode = types.MarketTradingModeNoTrading
-	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
-	m.stateChanged = true
 
-	if m.settlementPriceInMarket == nil {
-		m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+	if m.mkt.State != types.MarketStateProposed && m.mkt.State != types.MarketStatePending {
+		m.mkt.State = types.MarketStateTradingTerminated
+		m.mkt.TradingMode = types.MarketTradingModeNoTrading
+		m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
+		m.stateChanged = true
+
+		if m.settlementPriceInMarket == nil {
+			m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+			return
+		}
+		m.settlementPriceWithLock(ctx)
+	} else {
+		m.log.Debug("Market is not terminated before its enactment time")
 		return
 	}
```

Comment:
Seeing Barney's somment, it seems that in the cose termination is triggered before the end of the opening auction, the market should then be Cancelled. I suppose we should add the Cancelled type to the market state enum, and set it to Cancell in this else condition.

Reply:
@jeremyletang ,
Added changes to `tradingTerminated` step to set market state to "Canceled" and cancel all orders for each party:
https://github.com/vegaprotocol/vega/pull/5668/commits/2a27fdbb4a9172200b26f4fe28500fe548bd52d9#diff-1782f5982916957ff0e7e82b0c7130c6d4c8ab4eb2e06f5c7b5b4a5925e15bcfR3181

As well as handing if the market state is handled in the `OnTick` step in the execution engine:
https://github.com/vegaprotocol/vega/pull/5668/commits/2a27fdbb4a9172200b26f4fe28500fe548bd52d9#diff-dedf0633c5a9f40bc6bd64eda022d287b45a334e77b08ca25c307069777582b5R375
-> removing from list, sending to the list of proposals to be rejected